contracts/warp-controller/src/tests/execute/account/test_create_account.rs

```
@@ -108,3 +108,53 @@ fn test_create_account_exists() {
         ContractError::AccountAlreadyExists {}
     )
 }
+
+#[test]
+// Test the case when a warp account calls warp controller's create_account.
+// This is not allowed as of 12/2022.
+fn test_create_account_by_account() {
+    let mut deps = mock_dependencies();
+    let env = mock_env();
+    let info = mock_info("vlad", &vec![coin(100, "uluna")]);
+
+    let _instantiate_res = instantiate_warp(
+        deps.as_mut(),
+        env.clone(),
+        info.clone(),
+        Some(info.sender.to_string()),
+        Uint64::new(0),
+        Uint128::new(0),
+        Uint128::new(0),
+        Uint128::new(0),
+    )
+    .unwrap();
+
+    let (_create_account_res_first, reply_res_first) =
+        create_warp_account(&mut deps, env.clone(), info.clone(), Uint64::new(0));
+
+    // Get address of warp account just created and assign it as the sender of next create_account
+    let reply_res_first_clone = reply_res_first.unwrap().clone();
+    let _attr_action = reply_res_first_clone
+        .attributes
+        .iter()
+        .find(|attr| attr.key == "action" && attr.value == "save_account")
+        .ok_or_else(|| StdError::generic_err("cannot find `save_account` event"))
```

cannot find `save_account` **event**  or **attribute**? reply fn writes event but this is actually an attribute.

---

yeah let's change this to attribute

---

We also don't need to extract the action since we don't actually use that value

---

was trying to be strict so we fail early if action doesn't match, but i'm cool with dropping action.